src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -276,18 +302,48 @@ public Cci.IUnitReference GetUnit(EmitContext context)
 
         public string NamespaceName => string.Empty;
 
-        internal static string GenerateDataFieldName(ImmutableArray<byte> data)
+        /// <summary>Creates a name to use as the field for storing the specified data blob or array to cache such a blob.</summary>
+        /// <param name="data">The data that will be stored directly in the field or in an array cached in the field.</param>
+        /// <param name="alignment">The alignment of the data.</param>
+        /// <param name="elementType">NotPrimitive if this field will store the data directly; otherwise, the element primitive type if this is to be used for the array with that type.</param>
+        /// <returns></returns>
+        internal static string GenerateDataFieldName(ImmutableArray<byte> data, ushort alignment = 1, SpecialType elementType = SpecialType.None)
```

Comment:
>ushort alignment = 1, SpecialType elementType = SpecialType.None [](http://example.com/codeflow?start=80&length=64)

Having default values here is error prone, Consumer must provide explicit values at the call site.